.github/ci.yml

```
@@ -0,0 +1,40 @@
+name: CI
+
+on: [push, pull_request]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  lint:
+    # Avoid duplicate jobs on PR from a branch on the same repo
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.event.pull_request.base.repo.full_name
+
+    name: "Continuous Integration / lint"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - name: Build
+        run: cargo build --workspace --all-features
+      - name: cargo fmt
+        run: cargo fmt --all -- --check
+      - name: Clippy
+        run: cargo clippy --workspace --all-features -- -D warnings
+      - name: Build docs
+        run: cargo doc --document-private-items --workspace --all-features --no-deps
+    env:
+      RUSTDOCFLAGS: -Dwarnings
```

I know this is a copy of my veil code, but I think this should be indented, so the environment variable is only set for cargo-doc, since we don't want another step failing because docs emit a warning
```suggestion
      env:
        RUSTDOCFLAGS: -Dwarnings
```